blockchain/filedao/filedao_v2_util.go

```diff
@@ -178,12 +173,7 @@ func (fd *fileDAOv2) getBlock(height uint64) (*block.Block, error) {
 		return nil, db.ErrNotExist
 	}
 	// check whether block in staging buffer or not
-	storeKey := blockStoreKey(height, fd.header)
-	if storeKey >= fd.blkStore.Size() {
```

Comment:
root-cause of the bug:
the batch has not been written to DB yet, but `blkStore.Size()` already increased 1. This caused code go to L190 attempting to read the block from DB, which is not yet written to disk yet.